service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerSubmittedTest.java

```diff
@@ -74,88 +79,83 @@ void shouldReturnUnsuccessfulResponseWithMalformedData() {
 
     @Test
     void shouldBuildNotificationTemplatesWithCompleteValues() throws Exception {
-        Map<String, Object> completeHmctsParameters = getCompleteParameters()
-            .put("court", FAMILY_COURT)
-            .build();
+        SubmitCaseHmctsTemplate completeHmctsParameters = getCompleteParameters(new SubmitCaseHmctsTemplate());
+        completeHmctsParameters.setCourt(FAMILY_COURT);
 
-        Map<String, Object> completeCafcassParameters = getCompleteParameters()
-            .put("cafcass", CAFCASS_COURT)
-            .build();
+        SubmitCaseCafcassTemplate completeCafcassParameters = getCompleteParameters(new SubmitCaseCafcassTemplate());
+        completeCafcassParameters.setCafcass(CAFCASS_COURT);
 
         postSubmittedEvent("core-case-data-store-api/callback-request.json");
 
         verify(notificationClient).sendEmail(
             HMCTS_COURT_SUBMISSION_TEMPLATE,
             HMCTS_ADMIN_EMAIL,
-            completeHmctsParameters,
+            completeHmctsParameters.toMap(mapper),
             CASE_REFERENCE.toString());
 
         verify(notificationClient).sendEmail(
             CAFCASS_SUBMISSION_TEMPLATE,
             CAFCASS_EMAIL,
-            completeCafcassParameters,
+            completeCafcassParameters.toMap(mapper),
             CASE_REFERENCE.toString());
 
         verify(notificationClient, never()).sendEmail(
             HMCTS_COURT_SUBMISSION_TEMPLATE,
             "FamilyPublicLaw+ctsc@gmail.com",
-            completeHmctsParameters,
+            completeHmctsParameters.toMap(mapper),
             CASE_REFERENCE.toString());
     }
 
     @Test
     void shouldBuildNotificationTemplatesWithValuesMissingInCallback() throws Exception {
         CaseDetails caseDetails = enableSendToCtscOnCaseDetails(NO);
 
-        Map<String, Object> expectedHmctsParameters = getInCompleteParameters()
-            .put("court", FAMILY_COURT)
-            .build();
+        SubmitCaseHmctsTemplate expectedHmctsParameters = getIncompleteParameters(new SubmitCaseHmctsTemplate());
```

Comment:
Do you think it would be better to directly change to map here instead of doing the conversion twice in the verify? Same for cafcass one as well.

Reply:
Good idea but we set further properties on the returned templates which are not shared across templates.

Reply:
maybe not directly here but maybe a bit later. Just so that we don't have to call the `.toMap()` multiple times.

Reply:
in this test we are converting the HMCTS parameters to a map twice when we could just do it once